### PR TITLE
Add multiconfiguration support to jenkins scripts

### PR DESCRIPTION
**Pull request by akva2**
This adds multi-configuration support to the jenkins build scripts. It will be used to do mpi-enabled builds as well on jenkins. The mechanism is pretty simple; you have two environment variables BTYPES and CMAKE_TOOLCHAIN_FILES that is pre-filled by the user. Split and then put into an a-array instead of user supplying the a-array as those are a bugger on the command line. The toolchain files have 'full power', you can set compilers and options etc. (ie we can put -DUSE_MPI=1 in the toolchain file).

WIP, and includes PR #168 since i built on top of that.

@bska does this look acceptable ? should i remove the additional information?
Note that it will look slightly less clean in the output for downstreams (when some downstream is the main module), as i will have to clone opm-common first to obtain the helper scripts. No way around that.

### CODE REVIEW

**Comment by akva2** [comment]
>Repository revisions:
>        [main module] opm-common=master
>         [downstream] ert=master
>         [downstream] opm-parser=master
>         [downstream] opm-output=master
>         [downstream] opm-material=master
>         [downstream] opm-core=master
>         [downstream] opm-grid=master
>         [downstream] opm-simulators=master
>         [downstream] opm-upscaling=master
>         [downstream] ewoms=master
>Configurations to process:
>        serial=/home/akva/kode/toolchains/gcc-5.3-linux.cmake

**Comment by bska** [comment]
> does this look acceptable ?

Yes, I think so.  This is useful.  Just out of curiosity: Do you have an example that features the `[upstream]` tag too?  I'd be interested in seeing how that one aligns with the rest.

**Comment by akva2** [comment]
sorry haven't gotten further than this today. i'll update with an upstream entry once i have a module with upstreams. it should align though, there's extra spacing to align it just like downstream is aligned.

**Comment by bska** [comment]
> it should align though, there's extra spacing to align it just like downstream is aligned.

Okay, then I'm probably misreading the diff.  It looks like there's a single space in both `[upstream]` and `[downstream]` but the latter is two characters longer than the former.

**Comment by akva2** [comment]
right. let me rephrase. the intention was to right-array the ]. :)

**Comment by akva2** [comment]
```
Repository revisions:
             [upstream] opm-common=pull/169/merge
             [upstream] ert=master
             [upstream] opm-parser=master
        [main module] opm-output=master
         [downstream] opm-output=master
         [downstream] opm-core=master
         [downstream] opm-grid=master
         [downstream] opm-simulators=master
         [downstream] opm-upscaling=master
         [downstream] ewoms=master
Configurations to process:
        gcc-5.3-linux-mpi=/home/akva/kode/toolchains/gcc-5.3-linux-mpi.cmake
        gcc-5.3-linux=/home/akva/kode/toolchains/gcc-5.3-linux.cmake
```

@bska spacing fixed, how it looks now.

**Comment by bska** [comment]
> spacing fixed

Affirmative.

> how it looks now.

Much better!  I like it a lot and I think this should go in--possibly without the hard-coded `-j16` requests.

**Comment by akva2** [comment]
thanks for pointing those out, added while testing locally..

**Comment by akva2** [comment]
jenkins build this please

**Comment by akva2** [comment]
jenkins build this please

**Comment by akva2** [comment]
jenkins build this with downstreams serial please

**Comment by akva2** [comment]
jenkins build this with downstreams serial please
